extensions/sql/transfer-process-store/src/test/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/SqlTransferProcessStoreTest.java

```diff
@@ -0,0 +1,474 @@
+        leaseEntity(tpId, "someone");
+
+        t1.transitionInitial(); //modify
+
+        // leased by someone else -> throw exception
+        assertThatThrownBy(() -> store.update(t1)).hasRootCauseInstanceOf(IllegalStateException.class);
+
+    }
+
+    @Test
+    void delete() {
+        var t1 = createTransferProcess("id1");
+        store.create(t1);
+
+        store.delete("id1");
+        assertThat(count()).isEqualTo(0);
+        assertThat(store.findAll(QuerySpec.none())).isEmpty();
+    }
+
+    @Test
+    void delete_isLeasedBySelf_shouldThrowException() {
+        var t1 = createTransferProcess("id1");
+        store.create(t1);
+        leaseEntity(t1.getId(), CONNECTOR_NAME);
+
+
+        assertThatThrownBy(() -> store.delete("id1")).hasRootCauseInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    void delete_isLeasedByOther_shouldThrowException() {
+        var t1 = createTransferProcess("id1");
+        store.create(t1);
+
+        leaseEntity(t1.getId(), "someone-else");
+
+        assertThatThrownBy(() -> store.delete("id1")).hasRootCauseInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    void delete_notExist() {
+        store.delete("not-exist");
+        //no exception should be raised
+    }
+
+    @Test
+    void findAll_noQuerySpec() {
+        var all = IntStream.range(0, 10)
+                .mapToObj(i -> createTransferProcess("id" + i))
+                .peek(store::create)
+                .collect(Collectors.toList());
+
+        assertThat(store.findAll(QuerySpec.none())).containsExactlyInAnyOrderElementsOf(all);
+    }
+
+    @Test
+    void findAll_verifyPaging() {
+        IntStream.range(0, 10)
+                .mapToObj(i -> createTransferProcess(String.valueOf(i)))
+                .forEach(store::create);
+
+        var qs = QuerySpec.Builder.newInstance().limit(5).offset(3).build();
+        assertThat(store.findAll(qs)).hasSize(5)
+                .extracting(TransferProcess::getId)
+                .map(Integer::parseInt)
+                .allMatch(id -> id >= 3 && id < 8);
+    }
+
+    @Test
+    void findAll_verifyPaging_pageSizeLargerThanCollection() {
+
+        IntStream.range(0, 10)
+                .mapToObj(i -> createTransferProcess(String.valueOf(i)))
+                .forEach(store::create);
+
+        var qs = QuerySpec.Builder.newInstance().limit(20).offset(3).build();
+        assertThat(store.findAll(qs))
+                .hasSize(7)
+                .extracting(TransferProcess::getId)
+                .map(Integer::parseInt)
+                .allMatch(id -> id >= 3 && id < 10);
+    }
+
+    @Test
+    void findAll_verifyPaging_pageSizeOutsideCollection() {
+
+        IntStream.range(0, 10)
+                .mapToObj(i -> createTransferProcess(String.valueOf(i)))
+                .forEach(store::create);
+
+        var qs = QuerySpec.Builder.newInstance().limit(10).offset(12).build();
+        assertThat(store.findAll(qs)).isEmpty();
+
+    }
+
+    @Test
```

Comment:
I personally don't like these empty tests. I know they represent functionality that will be added later, but I'd rather have them deleted or marked for skipping.

Reply:
i don't either. i simply forgot to add them :/